File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::unordered_set<int> set {};
        for (auto &num : nums) {
            set.insert(num);
        }

        int max_seq = 0;
        int curr_seq = 0;
        for (auto &num : nums) {
            if (!set.count(num - 1)) {
                int curr_num = num;
                while(true) {
                    curr_seq += 1;
                    curr_num += 1;
                    if (!set.count(curr_num)) {
                        break;
                    }
                }
            }
            max_seq = std::max(max_seq, curr_seq);
            curr_seq = 0;
        } 

        return max_seq;

    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        max_count = ''
        count = {}
        l = 0
        r = 0
        max_length = 0

        while r < len(s):
            count[s[r]] = count.get(s[r], 0) + 1
            if count.get(max_count, 0) < count[s[r]]:
                max_count = s[r]
            
            while r - l - count[max_count] >= k:
                count[s[l]] -= 1
                prev = s[l]
                l += 1
                if count[max_count] == count[prev]:
                    max_count = s[l]
            r += 1
            max_length = max(max_length, r - l)
        
        return max_length